### DISCUSSION TITLE
n/a trip and missing drives, wrong segmentation of drives

### DISCUSSION THREAD

**Discussion by yumeng426** [teslamate-org/teslamate#1859] [Q&A]
Starting from the most recent update, many drives are missing in drive panel. If i check the trips, they are included in the N/A with empty start end point. If i click into it, it shows the trips from that date until now. Since I did not drive a lot, it seems only happen on long trips. 

example in trip panel

my actual trip starts from 11:23 am till Popeyes , but it splits to one starts at 11:23 am with -- start or anything and another starts in the middle of the trip. 
<img width="1662" alt="trip1" src="https://user-images.githubusercontent.com/11802847/126856003-ac0d2e19-6a34-42d5-9ccd-981e3dcc1c3e.png">

If i click at the wabash trip, it shows this, which is correct according to the start / end info. 
<img width="1706" alt="trip1-1" src="https://user-images.githubusercontent.com/11802847/126856002-1ea1c337-e86e-4e44-b87b-cd7846b0f77d.png">

if i click at the dashed one, which is what normally should be. it shows the trip from 11:23 am until now. The actual trip should be the segment circled in green. consumptions, distance and etc are shown as N/A
<img width="1672" alt="trip1-0" src="https://user-images.githubusercontent.com/11802847/126856051-b5c59764-23e7-43c0-9d1a-fe91c4a65dbb.png">

The segment circled in red in upper image is actually corresponding to the second trip after supercharge. but it is split to multiple empty ones and a short segment. each dashed option, if i click, shows the trips from that starting time till now. The actual trip is from 1:50 pm to u michigan, without stopping. 
<img width="1672" alt="trip2" src="https://user-images.githubusercontent.com/11802847/126856004-bd582017-7b26-4859-9898-f426139edde1.png">

**Comment by eigi60** [comment]
I have the same issue.
Any suggestion or solution available?
<img width="1144" alt="Bildschirmfoto 2021-08-17 um 20 38 29" src="https://user-images.githubusercontent.com/68687123/129782282-9f6bc6db-265c-46ba-a64b-3cb26f441793.png">
<img width="1150" alt="Bildschirmfoto 2021-08-17 um 20 38 52" src="https://user-images.githubusercontent.com/68687123/129782291-591d1b96-dc8d-4c45-8a1a-cbf306155362.png">

**Reply by DrMichael** [reply]
Could you dump the drives in that time slot out of the database?
`docker-compose exec database psql -U teslamate -d teslamate`
`SELECT * FROM drives WHERE start_date BETWEEN '2021-08-14T22:00:00Z' AND '2021-08-16T21:59:59Z';`

Or dump the data with "Inspect->Data", Data Options: Download for Excel, Download CSV.

**Reply by yumeng426** [reply]
I still have the same issue. I don't know how to fix it. I hope there is a way to fix it.

**Comment by eigi60** [comment]
Hi,

see attached the csv file.
Let me know if you need further information or data.

Regards
Christian

**Reply by DrMichael** [reply]
Am I blind or is there no attachment?

**Reply by eigi60** [reply]
[data-1629291671474.csv](https://github.com/adriankumpf/teslamate/files/7009752/data-1629291671474.csv)

**Comment by DrMichael** [comment]
No CSV attached?

**Comment by eigi60** [comment]


**Reply by DrMichael** [reply]
Hmmm, don't see it...

**Comment by eigi60** [comment]
can I upload it at my thread?


> Am 18.08.2021 um 20:06 schrieb Christian Eigner ***@***.***>:
> 
> ﻿lets try it again 😁
> in my mail it shows the attachment. 
> 
> hope you get it now
> 
> <data-1629291671474.csv>
> 
>> 
**

**Comment by eigi60** [comment]
here it is. Sorry thought when I answer to your mail, it will uploaded as well. 
[data-1629291671474.csv](https://github.com/adriankumpf/teslamate/files/7009744/data-1629291671474.csv)

**Comment by DrMichael** [comment]
Have there really been drives, where some are shown without start and end (and end_date...)?

So, the drives from the drives table in that time range would be intersting.

**Reply by eigi60** [reply]
the download is from the drives table. Or what do you mean?
The issue is, when I click on details for a drive without any start and end values, it shows the whole "daily" trip.

If the records without values are not considered in the statistics, then it's not a big deal, only a not nice visualization view.

**Reply by yumeng426** [reply]
Exactly same as mine.

**Comment by eigi60** [comment]
here an comparison from the drives between TeslaMate and Tronity. 
Take a look at the drives at 12:04 or 16:40.
In Teslamate it is a drive with "-" in Tronity are real data.
Could it be that 2 tracking tools making problems?
 
<img width="1123" alt="image" src="https://user-images.githubusercontent.com/68687123/130099741-cac02526-808e-443c-90e9-35c3e1238047.png">

<img width="990" alt="image" src="https://user-images.githubusercontent.com/68687123/130100259-e08f3f36-ecce-44bd-b6b0-ef3454b77bc2.png">

**Reply by eigi60** [reply]
some more information. It looks like that I have this issue since the last teslamate update. Before, I had always again some drives with missing consumtion value in the trip dashboard, but open detail of a drive, everything was ok. That happened only for short drives is the past. 
<img width="1149" alt="image" src="https://user-images.githubusercontent.com/68687123/130195698-e800d6c4-82cb-40e3-a414-90720977731f.png">

In drives, the records at 12:04 and 16:40 are not shown, in trip they are there with "-".
<img width="1170" alt="image" src="https://user-images.githubusercontent.com/68687123/130194943-71f87dd1-c90b-4dcf-992a-3a9dfad2b294.png">

in the Timeline Dashboard it looks the same as in Trip Dashboard.
<img width="1167" alt="image" src="https://user-images.githubusercontent.com/68687123/130195112-d4dad38c-f3f8-4b7f-a3c6-9ce538be16f7.png">

**Comment by jun3280net** [comment]
While mine is not nearly as bad as what has been identified by others, I can confirm this has been happening more frequently recently - which affects my Timeline View.  What's weird is that the start_date is populated with a value, but for whatever reason, a new drive record is created a few minutes later.   I end up deleting the bad records, but it's unclear why this is happening.

**Timeline View**
![image](https://user-images.githubusercontent.com/30555695/130336670-32a346d7-7cf3-4834-b2ac-d7ac0768d3a5.png)


**DRIVES Table**
![image](https://user-images.githubusercontent.com/30555695/130336575-e1ed7144-406c-42aa-bbe4-7dd05e9b3a55.png)

**Comment by DrMichael** [comment]
Hmmm, do we have teslamate logs for the respective times?

**Comment by eigi60** [comment]
I still have that issue from time to time. No clue when and why that happens.
Would really like to get that solved

<img width="615" alt="Bildschirmfoto 2021-10-29 um 09 29 01" src="https://user-images.githubusercontent.com/68687123/139394027-cb54854e-318b-4744-a7c1-63bb0345efd9.png">
.
